File: health-check.js
/**
 * Health Check Script for Docker Container
 * Performs health checks on the Executive Safety Dashboard application
 */

const http = require('http');
const process = require('process');

const HOST = process.env.HOST || 'localhost';
const PORT = process.env.PORT || 3000;
const TIMEOUT = parseInt(process.env.HEALTH_CHECK_TIMEOUT) || 5000;

/**
 * Performs HTTP health check
 */
function performHealthCheck() {
  return new Promise((resolve, reject) => {
    const options = {
      hostname: HOST,
      port: PORT,
      path: '/health',
      method: 'GET',
      timeout: TIMEOUT
    };

    const req = http.request(options, (res) => {
      let data = '';
      
      res.on('data', (chunk) => {
        data += chunk;
      });
      
      res.on('end', () => {
        if (res.statusCode === 200) {
          try {
            const response = JSON.parse(data);
            if (response.status === 'OK') {
              resolve(response);
            } else {
              reject(new Error(`Health check failed: ${response.status}`));
            }
          } catch (error) {
            reject(new Error(`Invalid JSON response: ${error.message}`));
          }
        } else {
          reject(new Error(`HTTP ${res.statusCode}: ${res.statusMessage}`));
        }
      });
    });

    req.on('error', (error) => {
      reject(new Error(`Request failed: ${error.message}`));
    });

    req.on('timeout', () => {
      req.destroy();
      reject(new Error(`Health check timeout after ${TIMEOUT}ms`));
    });

    req.setTimeout(TIMEOUT);
    req.end();
  });
}

/**
 * Main health check execution
 */
async function main() {
  try {
    console.log(`Performing health check on ${HOST}:${PORT}`);
    const result = await performHealthCheck();
    
    console.log('Health check passed:', {
      status: result.status,
      uptime: result.uptime,
      timestamp: result.timestamp
    });
    
    process.exit(0);
  } catch (error) {
    console.error('Health check failed:', error.message);
    process.exit(1);
  }
}

// Run health check if called directly
if (require.main === module) {
  main();
}

module.exports = { performHealthCheck };

File: analytics/health_check.py
"""
Health Check Script for Analytics Service Container
Performs health checks on the Python analytics service
"""

import asyncio
import sys
import os
import aiohttp
from datetime import datetime

# Configuration
HOST = os.getenv('HOST', 'localhost')
PORT = int(os.getenv('PORT', 8000))
TIMEOUT = int(os.getenv('HEALTH_CHECK_TIMEOUT', 5))

async def perform_health_check():
    """
    Performs async HTTP health check on the analytics service
    """
    url = f"http://{HOST}:{PORT}/health"
    
    try:
        timeout = aiohttp.ClientTimeout(total=TIMEOUT)
        async with aiohttp.ClientSession(timeout=timeout) as session:
            async with session.get(url) as response:
                if response.status == 200:
                    data = await response.json()
                    
                    if data.get('status') == 'healthy':
                        print(f"Health check passed: {data}")
                        return True
                    else:
                        print(f"Health check failed: Service unhealthy - {data}")
                        return False
                else:
                    print(f"Health check failed: HTTP {response.status}")
                    return False
                    
    except asyncio.TimeoutError:
        print(f"Health check failed: Timeout after {TIMEOUT}s")
        return False
    except aiohttp.ClientError as e:
        print(f"Health check failed: Client error - {e}")
        return False
    except Exception as e:
        print(f"Health check failed: Unexpected error - {e}")
        return False

async def main():
    """Main health check execution"""
    print(f"Performing health check on analytics service at {HOST}:{PORT}")
    print(f"Timestamp: {datetime.utcnow().isoformat()}")
    
    success = await perform_health_check()
    
    if success:
        print("Analytics service health check: PASSED")
        sys.exit(0)
    else:
        print("Analytics service health check: FAILED")
        sys.exit(1)

if __name__ == "__main__":
    asyncio.run(main())